### ISSUE TITLE
[Linux] Map Process not terminating reliably

### ISSUE THREAD

**Issue by CatsEyeXI** [DerpyProjectGroup/topaz#1819]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)

## Branch affected by issue 

`base`

## Steps to reproduce
make a few attempts at exiting the map server from the console.
![image](https://user-images.githubusercontent.com/65316697/170836819-30df3a28-1def-4c95-ac2a-1b7189229f54.png)

In the second test it went into a full loop and spammed my screen until i could open another tab and kill it.

## Expected behavior

Process is expected to exit cleanly so it can be managed by a restart script.

**Comment by zach2good** [comment]
For my own reference, threadsan catches this:
```
zach2good@ZACH-LAPTOP:/mnt/c/ffxi/server$ ./xi_map
[05/28/22 21:20:16:914][map][info][status] do_init: begin server initialization (do_init:180)
[05/28/22 21:20:16:914][map][info][status] The map-server is ready to work! (do_init:182)
[05/28/22 21:20:16:914][map][info][message] ======================================================================= (do_init:183)
[05/28/22 21:20:16:914][map][info][status] Console input thread is ready... (ConsoleService:67)
[05/28/22 21:20:16:914][map][info][status] Type 'help' for a list of available commands. (ConsoleService:68)
exit
> Goodbye!
==================
WARNING: ThreadSanitizer: data race (pid=10297)
  Read of size 8 at 0x0000010b7e18 by thread T3 (mutexes: write M451):
    #0 std::unique_ptr<ConsoleService, std::default_delete<ConsoleService> >::~unique_ptr() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:360:6 (xi_map+0x4d143d)
    #1 cxa_at_exit_wrapper(void*) <null> (xi_map+0x49cbff)
    #2 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_2&>(std::__invoke_other, ConsoleService::ConsoleService()::$_2&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f1e9f)
    #3 std::enable_if<is_invocable_r_v<void, ConsoleService::ConsoleService()::$_2&>, void>::type std::__invoke_r<void, ConsoleService::ConsoleService()::$_2&>(ConsoleService::ConsoleService()::$_2&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:111:2 (xi_map+0x4f1e7f)
    #4 std::_Function_handler<void (), ConsoleService::ConsoleService()::$_2>::_M_invoke(std::_Any_data const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:290:9 (xi_map+0x4f1d39)
    #5 std::function<void ()>::operator()() const /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:590:9 (xi_map+0x4efec1)
    #6 ConsoleService::MainLoop() /mnt/c/ffxi/server/src/common/console_service.cpp:131:13 (xi_map+0x4f25a5)
    #7 ConsoleService::ConsoleService()::$_3::operator()() const /mnt/c/ffxi/server/src/common/console_service.cpp:80:25 (xi_map+0x4f25a5)
    #8 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_3>(std::__invoke_other, ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f2143)
    #9 std::__invoke_result<ConsoleService::ConsoleService()::$_3>::type std::__invoke<ConsoleService::ConsoleService()::$_3>(ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (xi_map+0x4f2143)
    #10 void std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (xi_map+0x4f2143)
    #11 std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (xi_map+0x4f2143)
    #12 std::thread::_State_impl<std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (xi_map+0x4f2143)
    #13 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_3>(std::__invoke_other, ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f2143)
    #14 std::__invoke_result<ConsoleService::ConsoleService()::$_3>::type std::__invoke<ConsoleService::ConsoleService()::$_3>(ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (xi_map+0x4f2143)
    #15 void std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (xi_map+0x4f2143)
    #16 std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (xi_map+0x4f2143)
    #17 std::thread::_State_impl<std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (xi_map+0x4f2143)
    #18 <null> <null> (libstdc++.so.6+0xdc2c2)
    #19 <null> <null> (libstdc++.so.6+0xdc2c2)
    #20 <null> <null> (libstdc++.so.6+0xdc2c2)
    #21 <null> <null> (libstdc++.so.6+0xdc2c2)
    #22 <null> <null> (libstdc++.so.6+0xdc2c2)

  Previous write of size 8 at 0x0000010b7e18 by main thread:
    [failed to restore the stack]

  As if synchronized via sleep:
    #0 nanosleep <null> (xi_map+0x45c56d)
    #1 void std::this_thread::sleep_for<long, std::ratio<1l, 1000l> >(std::chrono::duration<long, std::ratio<1l, 1000l> > const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/this_thread_sleep.h:82:9 (xi_map+0x4f2f1a)
    #2 ConsoleService::ConsoleService()::$_3::operator()() const /mnt/c/ffxi/server/src/common/console_service.cpp:87:17 (xi_map+0x4f26f6)
    #3 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_3>(std::__invoke_other, ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f2143)
    #4 std::__invoke_result<ConsoleService::ConsoleService()::$_3>::type std::__invoke<ConsoleService::ConsoleService()::$_3>(ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (xi_map+0x4f2143)
    #5 void std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (xi_map+0x4f2143)
    #6 std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (xi_map+0x4f2143)
    #7 std::thread::_State_impl<std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (xi_map+0x4f2143)
    #8 <null> <null> (libstdc++.so.6+0xdc2c2)
    #9 <null> <null> (libstdc++.so.6+0xdc2c2)
    #10 <null> <null> (libstdc++.so.6+0xdc2c2)
    #11 <null> <null> (libstdc++.so.6+0xdc2c2)
    #12 <null> <null> (libstdc++.so.6+0xdc2c2)

  Location is global 'gConsoleService' of size 8 at 0x0000010b7e18 (xi_map+0x0000010b7e18)

  Mutex M451 (0x7b1c000001c8) created at:
    #0 pthread_mutex_lock <null> (xi_map+0x47c9c6)
    #1 __gthread_mutex_lock(pthread_mutex_t*) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11/bits/gthr-default.h:749:12 (xi_map+0x4f0b73)
    #2 std::mutex::lock() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_mutex.h:100:17 (xi_map+0x4f0b73)
    #3 std::lock_guard<std::mutex>::lock_guard(std::mutex&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_mutex.h:229:19 (xi_map+0x4ef8f6)
    #4 std::lock_guard<std::mutex>::lock_guard(std::mutex&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_mutex.h:229:19 (xi_map+0x4ef8f6)
    #5 ConsoleService::RegisterCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()>) /mnt/c/ffxi/server/src/common/console_service.cpp:105:21 (xi_map+0x4ef1c7)
    #6 ConsoleService::ConsoleService() /mnt/c/ffxi/server/src/common/console_service.cpp:41:5 (xi_map+0x4ee65f)
    #7 std::_MakeUniq<ConsoleService>::__single_object std::make_unique<ConsoleService>() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (xi_map+0x4d1346)
    #8 do_init(int, char**) /mnt/c/ffxi/server/src/map/map.cpp:185:23 (xi_map+0x520e30)
    #9 main /mnt/c/ffxi/server/src/common/kernel.cpp:265:5 (xi_map+0x4f6e13)

  Thread T3 (tid=10301, running) created by main thread at:
    #0 pthread_create <null> (xi_map+0x45ec1b)
    #1 std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) <null> (libstdc++.so.6+0xdc398)
    #2 std::_MakeUniq<ConsoleService>::__single_object std::make_unique<ConsoleService>() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (xi_map+0x4d1346)
    #3 std::_MakeUniq<ConsoleService>::__single_object std::make_unique<ConsoleService>() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (xi_map+0x4d1346)
    #4 do_init(int, char**) /mnt/c/ffxi/server/src/map/map.cpp:185:23 (xi_map+0x520e30)
    #5 main /mnt/c/ffxi/server/src/common/kernel.cpp:265:5 (xi_map+0x4f6e13)

SUMMARY: ThreadSanitizer: data race /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:360:6 in std::unique_ptr<ConsoleService, std::default_delete<ConsoleService> >::~unique_ptr()
==================
==================
WARNING: ThreadSanitizer: data race (pid=10297)
  Read of size 8 at 0x7b1c000001c0 by thread T3 (mutexes: write M451):
    #0 std::thread::joinable() const /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:176:16 (xi_map+0x4ef788)
    #1 std::thread::~thread() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:151:11 (xi_map+0x4ef788)
    #2 ConsoleService::~ConsoleService() /mnt/c/ffxi/server/src/common/console_service.cpp:99:1 (xi_map+0x4ef868)
    #3 ConsoleService::~ConsoleService() /mnt/c/ffxi/server/src/common/console_service.cpp:99:1 (xi_map+0x4ef868)
    #4 std::default_delete<ConsoleService>::operator()(ConsoleService*) const /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:85:2 (xi_map+0x4d1478)     
    #5 std::unique_ptr<ConsoleService, std::default_delete<ConsoleService> >::~unique_ptr() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:361:4 (xi_map+0x4d1478)
    #6 cxa_at_exit_wrapper(void*) <null> (xi_map+0x49cbff)
    #7 cxa_at_exit_wrapper(void*) <null> (xi_map+0x49cbff)
    #8 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_2&>(std::__invoke_other, ConsoleService::ConsoleService()::$_2&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f1e9f)
    #9 std::enable_if<is_invocable_r_v<void, ConsoleService::ConsoleService()::$_2&>, void>::type std::__invoke_r<void, ConsoleService::ConsoleService()::$_2&>(ConsoleService::ConsoleService()::$_2&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:111:2 (xi_map+0x4f1e7f)
    #10 std::_Function_handler<void (), ConsoleService::ConsoleService()::$_2>::_M_invoke(std::_Any_data const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:290:9 (xi_map+0x4f1d39)
    #11 std::function<void ()>::operator()() const /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:590:9 (xi_map+0x4efec1)
    #12 ConsoleService::MainLoop() /mnt/c/ffxi/server/src/common/console_service.cpp:131:13 (xi_map+0x4f25a5)
    #13 ConsoleService::ConsoleService()::$_3::operator()() const /mnt/c/ffxi/server/src/common/console_service.cpp:80:25 (xi_map+0x4f25a5)
    #14 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_3>(std::__invoke_other, ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f2143)
    #15 std::__invoke_result<ConsoleService::ConsoleService()::$_3>::type std::__invoke<ConsoleService::ConsoleService()::$_3>(ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (xi_map+0x4f2143)
    #16 void std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (xi_map+0x4f2143)
    #17 std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (xi_map+0x4f2143)
    #18 std::thread::_State_impl<std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (xi_map+0x4f2143)
    #19 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_3>(std::__invoke_other, ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f2143)
    #20 std::__invoke_result<ConsoleService::ConsoleService()::$_3>::type std::__invoke<ConsoleService::ConsoleService()::$_3>(ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (xi_map+0x4f2143)
    #21 void std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (xi_map+0x4f2143)
    #22 std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (xi_map+0x4f2143)
    #23 std::thread::_State_impl<std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (xi_map+0x4f2143)
    #24 <null> <null> (libstdc++.so.6+0xdc2c2)
    #25 <null> <null> (libstdc++.so.6+0xdc2c2)
    #26 <null> <null> (libstdc++.so.6+0xdc2c2)
    #27 <null> <null> (libstdc++.so.6+0xdc2c2)
    #28 <null> <null> (libstdc++.so.6+0xdc2c2)

  Previous write of size 8 at 0x7b1c000001c0 by main thread:
    [failed to restore the stack]

  As if synchronized via sleep:
    #0 nanosleep <null> (xi_map+0x45c56d)
    #1 void std::this_thread::sleep_for<long, std::ratio<1l, 1000l> >(std::chrono::duration<long, std::ratio<1l, 1000l> > const&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/this_thread_sleep.h:82:9 (xi_map+0x4f2f1a)
    #2 ConsoleService::ConsoleService()::$_3::operator()() const /mnt/c/ffxi/server/src/common/console_service.cpp:87:17 (xi_map+0x4f26f6)
    #3 void std::__invoke_impl<void, ConsoleService::ConsoleService()::$_3>(std::__invoke_other, ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:61:14 (xi_map+0x4f2143)
    #4 std::__invoke_result<ConsoleService::ConsoleService()::$_3>::type std::__invoke<ConsoleService::ConsoleService()::$_3>(ConsoleService::ConsoleService()::$_3&&) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/invoke.h:96:14 (xi_map+0x4f2143)
    #5 void std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::_M_invoke<0ul>(std::_Index_tuple<0ul>) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:253:13 (xi_map+0x4f2143)
    #6 std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> >::operator()() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:260:11 (xi_map+0x4f2143)
    #7 std::thread::_State_impl<std::thread::_Invoker<std::tuple<ConsoleService::ConsoleService()::$_3> > >::_M_run() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:211:13 (xi_map+0x4f2143)
    #8 <null> <null> (libstdc++.so.6+0xdc2c2)
    #9 <null> <null> (libstdc++.so.6+0xdc2c2)
    #10 <null> <null> (libstdc++.so.6+0xdc2c2)
    #11 <null> <null> (libstdc++.so.6+0xdc2c2)
    #12 <null> <null> (libstdc++.so.6+0xdc2c2)

  Location is heap block of size 112 at 0x7b1c000001c0 allocated by main thread:
    #0 operator new(unsigned long) <null> (xi_map+0x4d0026)
    #1 std::_MakeUniq<ConsoleService>::__single_object std::make_unique<ConsoleService>() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:30 (xi_map+0x4d133b)
    #2 do_init(int, char**) /mnt/c/ffxi/server/src/map/map.cpp:185:23 (xi_map+0x520e30)
    #3 main /mnt/c/ffxi/server/src/common/kernel.cpp:265:5 (xi_map+0x4f6e13)

  Mutex M451 (0x7b1c000001c8) created at:
    #0 pthread_mutex_lock <null> (xi_map+0x47c9c6)
    #1 __gthread_mutex_lock(pthread_mutex_t*) /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11/bits/gthr-default.h:749:12 (xi_map+0x4f0b73)

  Thread T3 (tid=10301, running) created by main thread at:
    #0 pthread_create <null> (xi_map+0x45ec1b)
    #1 std::thread::_M_start_thread(std::unique_ptr<std::thread::_State, std::default_delete<std::thread::_State> >, void (*)()) <null> (libstdc++.so.6+0xdc398)    #2 std::_MakeUniq<ConsoleService>::__single_object std::make_unique<ConsoleService>() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (xi_map+0x4d1346)
    #3 std::_MakeUniq<ConsoleService>::__single_object std::make_unique<ConsoleService>() /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/unique_ptr.h:962:34 (xi_map+0x4d1346)
    #4 do_init(int, char**) /mnt/c/ffxi/server/src/map/map.cpp:185:23 (xi_map+0x520e30)
    #5 main /mnt/c/ffxi/server/src/common/kernel.cpp:265:5 (xi_map+0x4f6e13)

SUMMARY: ThreadSanitizer: data race /usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_thread.h:176:16 in std::thread::joinable() const
==================
```

**Comment by WinterSolstice8** [comment]
Can confirm
![image](https://user-images.githubusercontent.com/60417494/170841177-3e0c0b7c-7bb1-40f9-b2f3-e9a8deabb5d3.png)


```
uname -a
Linux vps 5.15.0-1006-oracle #8-Ubuntu SMP Fri May 20 08:21:02 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux

gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/aarch64-linux-gnu/11/lto-wrapper
Target: aarch64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 11.2.0-19ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-11/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-11 --program-prefix=aarch64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libquadmath --disable-libquadmath-support --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --enable-fix-cortex-a53-843419 --disable-werror --enable-checking=release --build=aarch64-linux-gnu --host=aarch64-linux-gnu --target=aarch64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.2.0 (Ubuntu 11.2.0-19ubuntu1)

cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04 LTS"

```

**Comment by kaincenteno** [comment]
is this related?

https://github.com/LandSandBoat/server/issues/1489

**Comment by claywar** [comment]
This should be resolved by #1854 ; however, there may be an additional issue that needs to be fixed.
